app/controllers/application_controller.rb

```diff
@@ -1,10 +1,61 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
+    enable :sessions
+    set :session_secret, "my_application_secret"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+      erb :index
+  end
+
+  get '/signup' do
+      if !!session[:id]
```

Comment:
we should build out some helper methods in our `ApplicationController` that check for `current_user` and `logged_in?`

```ruby 

helpers do 

  def logged_in?
    !!current_user 
  end 

  def current_user 
    @current_user ||= User.find_by_id(session[:id])
  end 

end 
```

anytime we are now wanting access to the logged_in user just call `current_user` in your controllers and view(.erb).

And instead of checking session[:id] call `logged_in?`